### PR TITLE
fix relative ref with http ref bug

### PR DESCRIPTION
**Pull request by fujigon**
This patch fixes https://github.com/swagger-api/swagger-parser/issues/805.

### CODE REVIEW

File: modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java [hunk 1]
```diff
@@ -138,4 +140,23 @@ public void testNestedExternalRefs(@Injectable final Schema mockedModel){
     	assertTrue(testedOpenAPI.getComponents().getSchemas().get("Contact") != null);
     	assertTrue(testedOpenAPI.getComponents().getSchemas().get("Address") != null);
     }
+
+	@Test
+	public void testRelativeRefIncludingUrlRef(@Injectable final Schema mockedModel) {
```
Comment (jmini) [inline modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java]: This requires a server call to `https://raw.githubusercontent.com/`. Please mock it in the unit tests.

**Comment by jmini** [comment]
The change looks good to me, but your new unit test requires a server call to `https://raw.githubusercontent.com/`. Please mock it in the unit tests.

Add a new parameter: `@Mocked RemoteUrl remoteUrl` and the expectations at the beginning of your method:

```java
	        final String url = "https://raw.githubusercontent.com/swagger-api/swagger-parser/v2.0.2/modules/swagger-parser-v3/src/test/resources/relative/globals.yaml";
	        final String expectedResult = "components:\n" + 
	                "  schemas:\n" + 
	                "    link-object:\n" + 
	                "      type: object\n" + 
	                "      additionalProperties:\n" + 
	                "        \"$ref\": \"#/components/schemas/rel-data\"\n" + 
	                "    rel-data:\n" + 
	                "      type: object\n" + 
	                "      required:\n" + 
	                "      - href\n" + 
	                "      properties:\n" + 
	                "        href:\n" + 
	                "          type: string\n" + 
	                "        note:\n" + 
	                "          type: string\n" + 
	                "    result:\n" + 
	                "      type: object\n" + 
	                "      properties:\n" + 
	                "        name:\n" + 
	                "          type: string\n" + 
	                "        _links:\n" + 
	                "          \"$ref\": \"#/components/schemas/link-object\"\n" + 
	                "";
	        List<AuthorizationValue> auths = null;

	        new Expectations() {{
                    RemoteUrl.urlToString(url, auths);
	            times = 1;
	            result = expectedResult;
	        }};
```

---

Second question:
Is is use that the old behavior adding `./` (I guess that this was for relative url) is still present for relative url?

**Comment by fujigon** [comment]
Thanks for the review, and fine example to support this PR.
I reflected your review point.

**Comment by fujigon** [comment]
I added a test to confirm that "./" is still attached to relative local ref file.

**Comment by jmini** [comment]
I am not owner of this project, but this PR looks good to me.

Thank you a lot for this contribution

**Comment by gracekarina** [comment]
Thanks for this PR @fujigon

**Comment by douglasbgray** [comment]
@gracekarina I ran into this issue in the 2.0.2 release and was about to file an issue, when I see it has been fixed here. Do you have a timeline for the 2.0.3 release with this fix?

Thanks!

**Pull request merged**